.generator/templates/model/model.mustache

```diff
@@ -65,11 +67,22 @@ export class {{classname}} {{#parent}}extends {{{parent}}} {{/parent}}{
       if ([{{#enumVars}}{{{value}}}, {{/enumVars}}undefined].includes(data.{{{baseName}}})) {
           res.{{name}} = data.{{{baseName}}};
       } else {
-          throw TypeError(`invalid enum value ${ data.{{{baseName}}} } for {{baseName}}`);
+          console.error(`invalid enum value ${ data.{{{baseName}}} } for {{baseName}}`);
+          throw new ParseException("enum", `invalid enum value ${ data.{{{baseName}}} } for {{baseName}}`);
       }
         {{/allowableValues}}
         {{^allowableValues}}
-      res.{{{name}}} = ObjectSerializer.deserialize(data.{{{baseName}}}, "{{#isEnum}}{{{datatypeWithEnum}}}{{/isEnum}}{{^isEnum}}{{{dataType}}}{{/isEnum}}", "{{dataFormat}}")
+        try {
+          res.{{{name}}} = ObjectSerializer.deserialize(data.{{{baseName}}}, "{{#isEnum}}{{{datatypeWithEnum}}}{{/isEnum}}{{^isEnum}}{{{dataType}}}{{/isEnum}}", "{{dataFormat}}")
+        } catch(error) {
+            if (res?.unparsedObject !== undefined) {
+              res.unparsedObject["{{{name}}}"] = data.{{{baseName}}};
+            } else {
+              res.unparsedObject = {};
```

Comment:
You should push that line up, and only do `res.unparsedObject["{{{name}}}"] = data.{{{baseName}}};` once.

Reply:
It caused an error at compile time, because the unparsedObject attribute was defined as optional and wasn't initialized, so it could be undefined. I'll find a work around it.